Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift

```diff
@@ -62,32 +73,71 @@ internal final class AnalyticsProvider: AnalyticsProviderProtocol {
     ) {
         self.apiClient = apiClient
         self.configuration = configuration
-        self.uniqueAssetAPIClient = UniqueAssetAPIClient<CheckoutAttemptIdResponse>(apiClient: apiClient)
+        self.uniqueAssetAPIClient = UniqueAssetAPIClient<InitialAnalyticsResponse>(apiClient: apiClient)
     }
 
     // MARK: - Internal
-    
-    internal func fetchAndCacheCheckoutAttemptIdIfNeeded() {
-        fetchCheckoutAttemptId { _ in /* Do nothing, the point is to trigger the fetching and cache the value  */ }
-    }
 
-    internal func fetchCheckoutAttemptId(completion: @escaping (String?) -> Void) {
-        guard configuration.isEnabled else {
+    internal func sendInitialAnalytics(with flavor: TelemetryFlavor, additionalFields: AdditionalAnalyticsFields?) {
+        guard configuration.isEnabled, configuration.isTelemetryEnabled else {
             checkoutAttemptId = "do-not-track"
-            completion(checkoutAttemptId)
             return
         }
+        if case .dropInComponent = flavor { return }
+        
+        let telemetryData = TelemetryData(flavor: flavor,
+                                          additionalFields: additionalFields,
+                                          context: configuration.context)
 
-        let checkoutAttemptIdRequest = CheckoutAttemptIdRequest()
+        let initialAnalyticsRequest = InitialAnalyticsRequest(data: telemetryData)
 
-        uniqueAssetAPIClient.perform(checkoutAttemptIdRequest) { [weak self] result in
+        uniqueAssetAPIClient.perform(initialAnalyticsRequest) { [weak self] result in
             switch result {
             case let .success(response):
                 self?.checkoutAttemptId = response.identifier
-                completion(response.identifier)
             case .failure:
-                completion(nil)
+                self?.checkoutAttemptId = nil
             }
         }
     }
+    
+    internal func send(info: AdyenAnalytics.Info) {
+        infos.append(info)
+    }
+    
+    internal func send(log: AdyenAnalytics.Log) {
+        logs.append(log)
+    }
+    
+    internal func send(error: AdyenAnalytics.Error) {
+        errors.append(error)
+        sendAll()
+    }
+    
+    private func setupTimer() {
+        let timer = Timer(timeInterval: 10, target: self, selector: #selector(sendAll), userInfo: nil, repeats: true)
+        timer.tolerance = 1
+        RunLoop.current.add(timer, forMode: .common)
+    }
```

Comment:
Is this used?

Reply:
No not yet. As I wrote before these parts were not needed yet. I will work on them on next PR